### DISCUSSION TITLE
'predict' feature and multi-gpu

### DISCUSSION THREAD

**Discussion by gianscarpe** [Lightning-AI/pytorch-lightning#10212] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Dear All,
is it possible to get predict results onto a single gpu/thread when using 'trainer.predict' and multiple-gpus?
I couldn't figure it out!
Thansk

**Comment by rohitgr7** [comment]
I think it does return predictions from all the devices when you do
```py
predictions = trainer.predict(...)
```

**Comment by gianscarpe** [comment]
Hi @rohitgr7, I think in my particular case this does not happen because the model returns a (pickable) object instead of a tensor. Pytorch distributed utilities have a specific function for gathering object. Could it be the case?
